pkg/lib/cockpit-components-typeahead-select.tsx

```diff
@@ -425,7 +430,7 @@ export const TypeaheadSelectBase: React.FunctionComponent<TypeaheadSelectProps>
           aria-controls="select-typeahead-listbox"
                 />
                 <TextInputGroupUtilities
-                    {...(!(isFiltering && filterValue) && !selected ? { style: { display: 'none' } } : {})}
+                    {...(!(isFiltering && filterValue) && !(props.selected && onClearSelection) ? { style: { display: 'none' } } : {})}
```

Comment:
Also feels like a straight upstreamable fix?

Reply:
If it's a fix... I have to admit I am not super sure how the Clear button is supposed to behave... I'll submit it.